pkg/controller/secretsmanager/secret/setup.go

```diff
@@ -113,6 +119,86 @@ type hooks struct {
 	kube   client.Client
 }
 
+func (e *hooks) lateInitialize(cr *svcapitypes.SecretParameters, resp *svcsdk.DescribeSecretOutput) error {
+	payload, err := e.getPayload(context.TODO(), cr)
+	if err := client.IgnoreNotFound(err); err != nil {
+		return err
+	}
+
+	// If the K8s does not exist, create it with the data from AWS
+	if payload == nil {
+		req := &svcsdk.GetSecretValueInput{
+			SecretId: resp.ARN,
+		}
+		valueResp, err := e.client.GetSecretValueWithContext(context.TODO(), req)
+		if err != nil {
+			return errors.Wrap(err, errGetSecretValue)
+		}
+		ref, err := getSecretRef(cr)
+		if err != nil {
+			return err
+		}
+
+		sc := &corev1.Secret{}
+		sc.ObjectMeta.Name = ref.Name
+		sc.ObjectMeta.Namespace = ref.Namespace
+
+		data, err := getAWSSecretData(ref, valueResp)
+		if err != nil {
+			return errors.Wrap(err, errGetAWSSecretValue)
+		}
+		sc.Data = data
```

Comment:
```suggestion
	data, err := getAWSSecretData(ref, valueResp)
	if err != nil {
		return errors.Wrap(err, errGetAWSSecretValue)
	}
	sc := &corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:      ref.Name,
			Namespace: ref.Namespace,
		},
		Data: data,
	}
```
nitpick: easier to follow.